### PR TITLE
Fix unicode field names (again)

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

**Comment by cpcloud** [comment]
@llllllllll can you give this a quick review?

File: datashape/coretypes.py [hunk 1]
```diff
@@ -968,7 +968,35 @@ def __getitem__(self, types):
         if not isinstance(types, tuple):
             types = types,
 
-        return self(map(self._unpack_slice, types, range(len(types))))
+        return self(list(map(self._unpack_slice, types, range(len(types)))))
+
+
+def unify_name_types(names):
```
Comment (llllllllll) [inline datashape/coretypes.py]: could we conditionally define this function based on the version of python. This is just identity in python3 when str is unicode

Comment (cpcloud) [inline datashape/coretypes.py]: Sure

**Comment by llllllllll** [comment]
Is there a reason that we want to avoid having mixed type keys? In python2 string and unicode hash the same so our getitem will still work. I don't mind promoting all the keys, just wondering.

**Comment by cpcloud** [comment]
Pandas 0.17.x is now enforcing that columns all have the same type in their test functions and we use them all over the place so this is really just to prevent further headaches down the road

**Comment by llllllllll** [comment]
Ah. Looks good to me.

**Comment by llllllllll** [comment]
Oh, could you add a whatsnew entry

**Pull request merged**